go/cmd/dolt/cli/arg_parser_helpers.go

```diff
@@ -163,6 +164,7 @@ func CreateCheckoutArgParser() *argparser.ArgParser {
 	ap := argparser.NewArgParser()
 	ap.SupportsString(CheckoutCoBranch, "", "branch", "Create a new branch named {{.LessThan}}new_branch{{.GreaterThan}} and start it at {{.LessThan}}start_point{{.GreaterThan}}.")
 	ap.SupportsFlag(ForceFlag, "f", "If there is any changes in working set, the force flag will wipe out the current changes and checkout the new branch.")
+	ap.SupportsString(TrackFlag, "t", "direct|inherit", "When creating a new branch, set up 'upstream' configuration. Defaults to 'direct'.")
```

Comment:
Probably better to not document `inherit` if it's not supported yet, otherwise customers might read the documentation, see it, try to use it, and then be frustrated that it doesn't work after we advertised it in the docs. If it's not documented, it's less noise, and we can always go back and add it. 

If you think this will be a really fast follow and already know how to implement inherit, then I'm okay with leaving it in, but I think it's better to leave it out until we have support for it.